homeassistant/components/nextdns/sensor.py

```diff
@@ -0,0 +1,299 @@
+        key="ipv6_queries_ratio",
+        coordinator_type=ATTR_IP_VERSIONS,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        icon="mdi:ip",
+        name="{profile_name} IPv6 Queries Ratio",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    NextDnsSensorEntityDescription(
+        key="validated_queries",
+        coordinator_type=ATTR_DNSSEC,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        icon="mdi:lock-check",
+        name="{profile_name} DNSSEC Validated Queries",
+        native_unit_of_measurement="queries",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    NextDnsSensorEntityDescription(
+        key="not_validated_queries",
+        coordinator_type=ATTR_DNSSEC,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        icon="mdi:lock-alert",
+        name="{profile_name} DNSSEC Not Validated Queries",
+        native_unit_of_measurement="queries",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    NextDnsSensorEntityDescription(
+        key="validated_queries_ratio",
+        coordinator_type=ATTR_DNSSEC,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        icon="mdi:lock-check",
+        name="{profile_name} DNSSEC Validated Queries Ratio",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add a NextDNS entities from a config_entry."""
+    sensors: list[NextDnsSensor] = []
+    coordinators = hass.data[DOMAIN][entry.entry_id]
+
+    for description in SENSORS:
+        sensors.append(
+            NextDnsSensor(coordinators[description.coordinator_type], description)
+        )
+
+    async_add_entities(sensors)
+
+
+class NextDnsSensor(CoordinatorEntity, SensorEntity):
+    """Define an NextDNS sensor."""
+
+    coordinator: NextDnsUpdateCoordinator
+
+    def __init__(
+        self,
+        coordinator: NextDnsUpdateCoordinator,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator)
+        self._attr_device_info = coordinator.device_info
+        self._attr_unique_id = f"{coordinator.profile_id}_{description.key}"
+        self._attr_name = cast(str, description.name).format(
+            profile_name=coordinator.profile_name
+        )
+        self._attr_native_value = getattr(coordinator.data, description.key)
+        self.entity_description = description
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._attr_native_value = getattr(
```

Comment:
Could use a lambda in description instead

Reply:
I think I need help here, what exactly do you mean?

Reply:
I mean something like this https://github.com/home-assistant/core/blob/dev/homeassistant/components/sensibo/sensor.py#L241

Reply:
Smart, I will address all your comments in the new PR.